lib/GHCup/Utils.hs

```diff
@@ -819,23 +832,41 @@ unpackToDir dfp av = do
     | ".tar.bz2" `isSuffixOf` fn ->
       liftE (untar . BZip.decompress =<< rf av)
     | ".tar" `isSuffixOf` fn -> liftE (untar =<< rf av)
+#if defined(TAR)
+    | ".zip" `isSuffixOf` fn -> withArchive av (unpackInto dfp)
+#else
+    -- libarchive supports zip
     | ".zip" `isSuffixOf` fn -> liftE (untar =<< rf av)
+#endif
     | otherwise -> throwE $ UnknownArchive fn
 
 
 getArchiveFiles :: (MonadReader env m, HasLog env, MonadIO m, MonadThrow m)
                 => FilePath       -- ^ archive path
-                -> Excepts '[UnknownArchive
+                -> Excepts '[ UnknownArchive
                             , ArchiveResult
                             ] m [FilePath]
 getArchiveFiles av = do
   let fn = takeFileName av
+#if defined(TAR)
+  let entries :: Monad m => BL.ByteString -> Excepts '[ArchiveResult] m [FilePath]
+      entries =
+          lE @ArchiveResult
+          . Tar.foldEntries
+            (\e x -> fmap (Tar.entryPath e :) x)
```

Comment:
`entryPath` is misleading: it's a low-level operation which does not account for long file names. You probably want to use [`decodeLongNames`](https://hackage.haskell.org/package/tar-0.6.0.0/docs/Codec-Archive-Tar-LongNames.html#v:decodeLongNames) followed by `Tar.foldEntries` with [`entryTarPath`](https://hackage.haskell.org/package/tar-0.6.0.0/docs/Codec-Archive-Tar-Types.html#t:GenEntry).

Reply:
Oops, you're right. I forgot when restoring the old commit.

Reply:
Now I'm worried... we changed the API so much that this might affect other users too. And there's no clear migration guide. The documentation doesn't warn either and no deprecation warning.

Reply:
Just updated the documentation, will include in the next release.